lib/rspec/matchers/built_in/change.rb

```diff
@@ -372,10 +372,15 @@ def evaluate_value_proc
         end
 
         def inspect_message_with_receiver(message, receiver)
-          if receiver.class == Class
+          case receiver
+          when Module
             "#{receiver}.#{message}"
+          when NilClass
+            "nil##{message}"
```

Comment:
I think the special notation `nil#some_message` is a bit odd. Isn't `NilClass#some_message` good enough?

If you did this to handle the following issue in the class extraction:

```ruby
[1] pry(main)> receiver = nil
=> nil
[2] pry(main)> singleton_class = class << receiver; self; end
=> NilClass
[3] pry(main)> klass = singleton_class.ancestors.find { |ancestor| !ancestor.equal?(singleton_class) }
=> Object # This is wrong
```

... we can simplify it as follows:

```ruby
def inspect_message_with_receiver(message, receiver)
  case receiver
  when Module
    "#{receiver}.#{message}"
  else
    "#{class_of(receiver)}##{message}"
  end
end

def class_of(object)
  object.class
rescue NoMethodError
  singleton_class = class << object; self; end
  singleton_class.ancestors.find { |ancestor| !ancestor.equal?(singleton_class) }
end
```